src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -282,7 +283,14 @@ class MarathonSchedulerActor private (
 
   def deploymentFailed(plan: DeploymentPlan, reason: Throwable): Unit = {
     logger.error(s"Deployment ${plan.id}:${plan.version} of ${plan.targetIdsString} failed", reason)
-    Future.sequence(plan.affectedRunSpecIds.map(launchQueue.purge))
+    Future.sequence(plan.affectedRunSpecIds.map { id =>
+      instanceTracker.specInstances(id).map { instances =>
```

Comment:
I'm not sure you should kill all instances. `LaunchQueue.purge` just removes all scheduled instances.

Should a failed deployment kill *all* instances of an app no matter the version?

Reply:
@zen-dog

Reply:
No, you don't have to kill any tasks here.